server/NucleiSegmentation/NucleiSegmentation.py

```diff
@@ -17,15 +18,33 @@
 }
 
 
+def read_image(file):
+    return skimage.io.imread(file)[:, :, :3]
+
+
 def main(args):
 
     #
     # Read Input Image
     #
     print('>> Reading input image')
 
-    imInput = skimage.io.imread(args.inputImageFile)[:, :, :3]
-
+    imInput = None
+    imageFile = args.inputImageFile
+    if os.path.isdir(imageFile):
+        dir = imageFile
+        for file in os.listdir(dir):
```

Comment:
In the case of dicom images, i am guessing there will be an item on girder with all files of the dicom series. In the case ITK, we just give the path of the .mhd file to itkImageFileReader.

In the case of other formats such as nrrd or png all the image information is within one file.